releasenotes-builder/src/main/java/com/github/checkstyle/MainProcess.java

```diff
@@ -79,7 +80,50 @@ public static List<String> runPostGenerationAndPublication(
             Multimap<String, ReleaseNotesMessage> releaseNotes, CliOptions cliOptions)
             throws IOException, TemplateException {
         runPostGeneration(releaseNotes, cliOptions);
-        return runPostPublication(cliOptions);
+        final List<String> errors = new ArrayList<>();
+        errors.addAll(validateNotes(releaseNotes, cliOptions));
+        errors.addAll(runPostPublication(cliOptions));
+        return errors;
+    }
+
+    /**
+     * Validate that pom version matches issues labels in release notes.
+     *
+     * @param releaseNotes map of release notes messages.
+     * @param cliOptions command line options.
+     * @return list of notes validation errors.
+     */
+    private static List<String> validateNotes(Multimap<String,
+                                                    ReleaseNotesMessage> releaseNotes,
+                                                    CliOptions cliOptions) {
+        final String releaseVersion = cliOptions.getReleaseNumber();
+        final long amountOfCommas = getAmountOfCommasInReleaseVersion(releaseVersion);
+
+        final boolean containsNewOrBreakingCompatabilityLabel =
+            releaseNotes.containsKey(Constants.NEW_FEATURE_LABEL)
+            || releaseNotes.containsKey(Constants.NEW_MODULE_LABEL)
+            || releaseNotes.containsKey(Constants.BREAKING_COMPATIBILITY_LABEL);
+
+        final List<String> errors = new ArrayList<>();
+
+        if (amountOfCommas == 1 && containsNewOrBreakingCompatabilityLabel
+            || amountOfCommas == 2 && !containsNewOrBreakingCompatabilityLabel) {
```

Comment:
Can we make methods like isMinor isPatch to explain magic numbers by method names

Reply:
Done.